### ISSUE TITLE
Sharing Dataset

### ISSUE THREAD

**Issue by akinsevimli** [SudhuRao/PointNet_radar#1]
Hello,

If it is possible, could you share the dataset?

Thank you.

**Comment by SudhuRao** [comment]
Hi..  The dataset was from the company and would not be possible to share. 

The kitti dataset has RADAR data in it. However it has more than 1 (I think 4) RADAR data.

The pointnet with reduced number of points did not produce a great shape of object. However it was able to classify well to an extent. It was also very susceptible to noisy data .

Hope this helps.

**Comment by akinsevimli** [comment]
Hi.
Thank you for then answer. 
I guess the shape of the data is not balanced. So, you added dummy points? Is this a problem for training?
Why didnt you consider RCS and Velocity values?

**Comment by akinsevimli** [comment]
By the way, what was the accuracy and loss results? How about the overfitting?

**Comment by SudhuRao** [comment]
Yes.. the shape of the dataset was not balanced. I had the radar images of vehicles and then some things like coke can on the road. 

The dimension of the object actually matters a lot. If I remember the paper correctly, the original implementation has toys in it with similar sized object.

Regarding dummy points, yes and no. I did not add the dummy points. The pointnet tries to get the core object points which defines the shape of the object (points which in any transformations holds identity). However the pointnet needs minimum number of points. The RADAR provides every 100ms around 4-5 points of the object. I collect this over time and transofrm it to a  dimension defined by me. I had defined the point the object is tracked at 4m front of the car, at a height equal to height of the radar camera placement.  Hence if the object which the radar is tracking (lets say car) has less then minimum number of points, then I was repeating the points(dummy) at the origin. This does not add any more information.

I did not add the velocity values as the objects that I was classifying, like car was always stationary which was not useful. Hence made the focus of the project to classify(or to check the possibility) based on shape.

Accuracy was good. But that was mainly due to the fact that dataset was easy. Not much to take home in that aspect.
However, I found it very sensitive to noise. The method of filling up the data (if points < minimum number of points) impacts the performance.

Hope this was helpful.

**Comment by akinsevimli** [comment]
Thank you for your reply.

I understood that you transformed all classes and of course the points to x =4, y =0 then you added the required size of points (4,0, 0) next to them??? 

All these dummy points could be overfitting? Because we use these points all the time. The model can memorize them. Otherwise, the transformed points will  include the dummy points, so the overfitting problem is not supposed to occur?

Have you tried to train with PointNet one by one ? By doing this, there is no need to make all data sample equal.
